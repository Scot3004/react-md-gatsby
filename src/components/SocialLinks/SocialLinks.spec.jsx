import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SocialLinks, {shareCount} from './SocialLinks';

const postNode = {
  "fields": {
    "slug": "/introducing-gatsby-material-starter"
  },
  "excerpt": "This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to vâ€¦",
  "timeToRead": 2,
  "frontmatter": {
    "title": "Introducing Gatsby Material Starter",
    "tags": [
      "gatsby",
      "material",
      "starter"
    ],
    "cover": "/logos/logo-1024.png",
    "date": "18/07/2017"
  }
}

it('renders correctly', () => {
  const tree = shallow(<SocialLinks postNode={postNode} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly mobile', () => {
  const tree = shallow(<SocialLinks postNode={postNode} mobile />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('share count 0', () => {
  const tree = shallow(shareCount(0));
  expect(toJson(tree)).toMatchSnapshot();
});

it('share count 0', () => {
  const tree = shallow(shareCount(10));
  expect(toJson(tree)).toMatchSnapshot();
});
