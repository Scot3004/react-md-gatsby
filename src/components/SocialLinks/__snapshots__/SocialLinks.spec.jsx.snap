// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="social-links"
>
  <CreatedButton
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={460}
    windowWidth={660}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
  </CreatedButton>
  <CreatedButton
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    quote="This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…"
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    description="This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…"
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={600}
    windowWidth={750}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={48}
    />
  </CreatedButton>
</div>
`;

exports[`renders correctly mobile 1`] = `
<div
  className="social-links"
>
  <CreatedButton
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={460}
    windowWidth={660}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
  </CreatedButton>
  <CreatedButton
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    quote="This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…"
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    description="This is a original post of ruben, but reflect all technologies applied on this website Not so long ago, this project got ported to v…"
    title="Introducing Gatsby Material Starter"
    url="https://www.secorto.com/undefined"
    windowHeight={600}
    windowWidth={750}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
    <Component
      url="https://www.secorto.com/undefined"
    />
  </CreatedButton>
  <CreatedButton
    url="https://www.secorto.com/undefined"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={36}
    />
  </CreatedButton>
</div>
`;

exports[`share count 0 1`] = `
<div
  className="share-count"
/>
`;

exports[`share count 0 2`] = `
<div
  className="share-count"
>
  10
</div>
`;
